Controller/NotifyHandler.php

```diff
@@ -195,12 +196,43 @@ private function _onNotificationSuccess($oResponse, $oBackendService, $sPaymentI
         }
 
         $oOrder = oxNew(Order::class);
+        $bSavedTransaction = self::_saveIfUnmatchedPoiPiaTransaction(
+            $sTransactionId,
+            $oResponse,
+            $oBackendService,
+            $oOrder
+        );
+
+        if (!$bSavedTransaction) {
+            ResponseHandler::onSuccessResponse($oResponse, $oBackendService, $oOrder);
+        }
+        return;
+    }
+
+    /**
+     * Saves POI/PIA transaction if it is unmatched, and returns true. Otherwise returns false.
+     *
+     * @param string          $sTransactionId
+     * @param SuccessResponse $oResponse
+     * @param BackendService  $oBackendService
+     * @param Order           $oOrder
+     *
+     * @return boolean
+     *
+     * @since 1.3.0
+     */
+    private function _saveIfUnmatchedPoiPiaTransaction($sTransactionId, $oResponse, $oBackendService, &$oOrder)
```

Comment:
Please rename to `_handleUnmatchedTransactions` as it is not only for POI/PIA (adjust description accordingly).

Reply:
Done.